utils/protocol-types-generator/index.js

```diff
@@ -72,12 +73,47 @@ declare global {
 
 export default Protocol;
 `;
-  const outputPath = path.join(__dirname, '..', '..', 'src', 'protocol.d.ts');
-  require('fs').writeFileSync(outputPath, output);
-  console.log(`Wrote protocol.d.ts for ${version} to ${path.relative(process.cwd(), outputPath)}`);
-  console.log(`You should commit the changes.`);
+
+  return {output, version};
 });
 
+const protocolOutputPath = path.join(__dirname, '..', '..', 'src', 'protocol.d.ts');
+const relativeProtocolOutputPath = path.relative(process.cwd(), protocolOutputPath);
+
+const writeOutputToDisk = ({output, version}) => {
+  require('fs').writeFileSync(protocolOutputPath, output);
+  console.log(`Wrote protocol.d.ts for ${version} to ${relativeProtocolOutputPath}`);
+  console.log(`You should commit the changes.`);
+};
+
+const cli = async() => {
+  const scriptToRun = process.argv[2];
+
+  if (scriptToRun === 'update') {
+    writeOutputToDisk(await fetchAndGenerateProtocolDefinitions());
+  } else if (scriptToRun === 'compare') {
+    const {output} = await fetchAndGenerateProtocolDefinitions();
+    const outputOnDisk = require('fs').readFileSync(protocolOutputPath, {encoding: 'utf8'});
+    if (output === outputOnDisk) {
+      console.log(`Success: ${relativeProtocolOutputPath} is up to date.`);
+    } else {
+      console.log(`Error: ${relativeProtocolOutputPath} is out of date.`);
+      console.log('You should run \`npm run update-protocol-d-ts\` and commit the changes.');
```

Comment:
unnecessary escaping:

```suggestion
      console.log('You should run `npm run update-protocol-d-ts` and commit the changes.');
```